lib/transproc/hash.rb

```diff
@@ -280,5 +280,37 @@ def unwrap!(hash, root, keys = nil)
 
       hash
     end
+
+    # Folds array of tuples to array of values from a specified key
+    #
+    # The second argument defines the key to fold tuples by
+    #
+    # @example
+    #   source = {
+    #     name: "Jane",
+    #     tasks: [{ title: "be nice", priority: 1 }, { title: "sleep well" }]
+    #   }
+    #   Transproc(:fold, :tasks, :title)[source]
+    #   # => { name: "Jane", tasks: ["be nice", "sleep well"] }
+    #   Transproc(:fold, :tasks, :priority)[source]
+    #   # => { name: "Jane", tasks: [1, nil] }
+    #
+    # @param [Hash]
+    #
+    # @return [Hash]
+    #
+    # @api public
+    def fold(hash, key, tuple_key)
+      hash.merge key => Transproc(:extract_key, tuple_key)[hash[key]]
```

Comment:
You can just `require 'transproc/array'` at the top of the file, if it's dependent on a transformation from another file, should do that